server/channels/app/imaging/decode.go

```diff
@@ -115,7 +115,11 @@ func (d *Decoder) DecodeConfig(rd io.Reader) (image.Config, string, error) {
 func GetDimensions(imageData io.Reader) (int, int, error) {
 	cfg, _, err := image.DecodeConfig(imageData)
 	if seeker, ok := imageData.(io.Seeker); ok {
-		defer seeker.Seek(0, 0)
+		defer func() {
+			if _, err = seeker.Seek(0, 0); err != nil {
+				fmt.Printf("warning: failed to reset reader position: %v\n", err)
+			}
+		}()
 	}
 	return cfg.Width, cfg.Height, err
 }
```

Comment:
Logging the error makes it hard to spot. Can we instead return it? Do we even need the `defer` clause?

I was wondering if code like this would make it easier to check both errors. What do you think?

```golang
func GetDimensions(imageData io.Reader) (width int, height int, err error) {
	cfg, _, err := image.DecodeConfig(imageData)
	width, height = cfg.Width, cfg.Height
	if seeker, ok := imageData.(io.Seeker); ok {
		_, err2 := seeker.Seek(0, 0)
		// Prefer the error from decoding the image config.
		if err == nil && err2 != nil {
			err = fmt.Errorf("failed to seek back to the beginning of the image data: %w", err2)
		}
	}

	return
}
```

Reply:
@Aryakoste Do you have questions about the feedback?

Reply:
yeah the defer isnt necessary here and your code makes it easier to check both errors. I have made the changes @hanzei